miio/integrations/waterheater/viomi/viomiwaterheater.py

```diff
@@ -0,0 +1,271 @@
+
+    @command(
+        click.argument("temperature", type=int),
+        default_output=format_output("Setting target temperature to {temperature}"),
+    )
+    def set_target_temperature(self, temperature: int):
+        """Set target temperature."""
+        min_temp: int
+        max_temp: int
+
+        min_temp, max_temp = SUPPORTED_MODELS.get(
+            self.model, SUPPORTED_MODELS[MODEL_WATERHEATER_E1]
+        )["temperature_range"]
+
+        if not min_temp <= temperature <= max_temp:
+            raise ViomiWaterHeaterException(
+                "Invalid target temperature: %s" % temperature
+            )
+        return self.send("set_temp", [temperature])
+
+    @command(default_output=format_output("Setting bacteriostatic mode on"))
+    def set_bacteriostatic_mode(self):
+        """Set bacteriostatic operational mode (water disinfection mode)."""
+        mode: OperationMode
+        bacteriostatic_mode: bool
+        bacteriostatic_temp: int
+
+        bacteriostatic_mode = SUPPORTED_MODELS.get(
+            self.model, SUPPORTED_MODELS[MODEL_WATERHEATER_E1]
+        )["bacteriostatic_mode"]
+
+        if not bacteriostatic_mode:
+            raise ViomiWaterHeaterException("Bacteriostatic mode is not supported.")
+
+        bacteriostatic_temp = SUPPORTED_MODELS.get(
+            self.model, SUPPORTED_MODELS[MODEL_WATERHEATER_E1]
+        )["bacteriostatic_temperature"]
+
+        mode = OperationMode(self.send("get_prop", ["modeType"])[0])
+
+        # No Bacteriostatic operational mode in Thermostatic mode.
+        if mode == OperationMode.Thermostatic:
+            raise ViomiWaterHeaterException(
+                "Bacteriostatic operational mode is "
+                "not supported in Thermostatic mode."
+            )
+        else:
+            return self.send("set_temp", [bacteriostatic_temp])
+
+    @command(
+        click.argument("booking_time_start", type=int),
+        click.argument("booking_time_end", type=int),
+        default_output=format_output(
+            lambda booking_time_start, booking_time_end: "Setting up the Booking mode operational interval from: %s "
+            % booking_time_start
+            + "to: %s " % booking_time_end
+            + "(duration: %s)"
+            % (
+                booking_time_end - booking_time_start
+                if booking_time_end - booking_time_start > 0
+                else booking_time_end - booking_time_start + 24
+            )
+        ),
+    )
+    def set_appoint(self, booking_time_start, booking_time_end):
+        """Setting up the Booking mode operational interval."""
+        if not (0 <= booking_time_start <= 23) or not (0 <= booking_time_end <= 23):
+            raise ViomiWaterHeaterException(
+                "Booking mode operational interval parameters "
+                "must be within [0, 23]."
+            )
+        else:
+            return self.send("set_appoint", [1, booking_time_start, booking_time_end])
```

Comment:
```suggestion
        return self.send("set_appoint", [1, booking_time_start, booking_time_end])
```
Please remove also other unnecessary elses. The error conditions should be checked as early as possible, and there is no need to perform any checks for the call itself when the preconditions are already checked.